src/proxy/EntryListener.ts

```diff
@@ -31,7 +31,7 @@ export interface EntryListener<K, V> {
     mapEvicted?: MapEventListener<K, V>;
     mapCleared?: MapEventListener<K, V>;
 
-    [event: string]: EntryEventListener<K, V> | MapEventListener<K, V>;
+    [event: string]: EntryEventListener<K, V> | MapEventListener<K, V> | undefined;
```

Comment:
Why do we need this line? I mean, aren't all possible events defined as properties above?

Reply:
It is defined but they can be undefined. This line is the general definition. It means any key can be mapped to either EntryEventListener, MapEventListener or undefined. A user mentioned if we don't put undefined here, typescript compilation fails for him.

Reply:
We may not need this line at all. Let me think

Reply:
We need this line in order to be able to index the type like `entryListener[stringVariable]`

Reply:
Yes, removing this line may make the code more complex because we will need more switch/case blocks.